lib/ansible/modules/network/nxos/nxos_interface.py

```diff
@@ -701,7 +701,7 @@ def main():
                 end_state, is_default = smart_existing(module, intf_type,
                                                        normalized_interface)
             else:
-                end_state = get_interfaces_dict(module)[interface_type]
+                get_interfaces_dict(module)[interface_type]
```

Comment:
This function doesn't have any side effects, you should be able to just kill the whole else block.

Reply:
Is this what you mean?
```python
            if module.params['interface']:
                if delta.get('mode'):
                    # if the mode changes from L2 to L3, the admin state
                    # seems to change after the API call, so adding a second API
                    # call to ensure it's in the desired state.
                    admin_state = delta.get('admin_state') or admin_state
                    c1 = 'interface {0}'.format(normalized_interface)
                    c2 = get_admin_state(delta, normalized_interface, admin_state)
                    cmds2 = [c1, c2]
                    load_config(module, cmds2)
                    cmds.extend(cmds2)
                end_state, is_default = smart_existing(module, intf_type,
                                                       normalized_interface)
            get_interfaces_dict(module)[interface_type]
            cmds = [cmd for cmd in cmds if cmd != 'configure']
```